### PR TITLE
SimplePie 1.5.8

### PR DESCRIPTION
**Pull request by math-GH**
SimplePie upgrades to 1.5.8

I guess it could be important because @Alkarex made some PRs.

On some points we have to take a closer look, because there were some lines of code marked with `// FreshRSS`

### CODE REVIEW

File: lib/SimplePie/SimplePie.php [hunk 1]
```diff
@@ -74,12 +74,6 @@
  */
 define('SIMPLEPIE_LINKBACK', '<a href="' . SIMPLEPIE_URL . '" title="' . SIMPLEPIE_NAME . ' ' . SIMPLEPIE_VERSION . '">' . SIMPLEPIE_NAME . '</a>');
 
-/**
- * Use syslog to report HTTP requests done by SimplePie.
- * @see SimplePie::set_syslog()
- */
-define('SIMPLEPIE_SYSLOG', true);	//FreshRSS
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:81]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 2]
```diff
@@ -1253,14 +1241,6 @@ public function add_attributes($attribs = '')
 		$this->sanitize->add_attributes($attribs);
 	}
 
-	/**
-	 * Use syslog to report HTTP requests done by SimplePie.
-	 */
-	public function set_syslog($value = SIMPLEPIE_SYSLOG)	//FreshRSS
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1259]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 3]
```diff
@@ -1518,21 +1471,15 @@ public function init()
 				{
 					$encodings[] = strtoupper($charset[1]);
 				}
-				else
-				{
-					$encodings[] = '';	//FreshRSS: Let the DOM parser decide first
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1523]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 4]
```diff
@@ -1518,21 +1471,15 @@ public function init()
...
 			}
 			elseif (in_array($sniffed, $text_types) || substr($sniffed, 0, 5) === 'text/' && substr($sniffed, -4) === '+xml')
 			{
 				if (isset($headers['content-type']) && preg_match('/;\x20?charset=([^;]*)/i', $headers['content-type'], $charset))
 				{
 					$encodings[] = strtoupper($charset[1]);
 				}
-				else
-				{
-					$encodings[] = '';	//FreshRSS: Let the DOM parser decide first
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1534]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 5]
```diff
@@ -1554,14 +1501,13 @@ public function init()
 		foreach ($encodings as $encoding)
 		{
 			// Change the encoding to UTF-8 (as we always use UTF-8 internally)
-			if ($utf8_data = (empty($encoding) || $encoding === 'UTF-8') ? $this->raw_data :	//FreshRSS
-				$this->registry->call('Misc', 'change_encoding', array($this->raw_data, $encoding, 'UTF-8')))
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1558]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 6]
```diff
@@ -1554,14 +1501,13 @@ public function init()
...
+			if ($utf8_data = $this->registry->call('Misc', 'change_encoding', array($this->raw_data, $encoding, 'UTF-8')))
 			{
 				// Create new parser
 				$parser = $this->registry->create('Parser');
 
 				// If it's parsed fine
-				if ($parser->parse($utf8_data, empty($encoding) ? '' : 'UTF-8', $this->permanent_url))	//FreshRSS
+				if ($parser->parse($utf8_data, 'UTF-8', $this->permanent_url))
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1510]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie.php [hunk 7]
```diff
@@ -1834,7 +1775,6 @@ protected function fetch_data(&$cache)
 			$locate = null;
 		}
 
-		$file->body = trim($file->body);	//FreshRSS
```
Comment (math-GH) [inline lib/SimplePie/SimplePie.php:1837]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie/Content/Type/Sniffer.php [hunk 8]
```diff
@@ -121,9 +123,7 @@ public function get_type()
 
 				return $official;
 			}
-			elseif ($official === 'text/html'
-				|| $official === 'text/xml'	//FreshRSS
-				|| $official === 'application/xml')	//FreshRSS
+			elseif ($official === 'text/html')
```
Comment (math-GH) [inline lib/SimplePie/SimplePie/Content/Type/Sniffer.php:126]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie/File.php [hunk 9]
```diff
@@ -78,10 +78,6 @@ public function __construct($url, $timeout = 10, $redirects = 5, $headers = null
 		$this->useragent = $useragent;
 		if (preg_match('/^http(s)?:\/\//i', $url))
 		{
-			if ($syslog_enabled)
-			{
-				syslog(LOG_INFO, 'SimplePie GET ' . SimplePie_Misc::url_remove_credentials($url));	//FreshRSS
-			}
```
Comment (math-GH) [inline lib/SimplePie/SimplePie/File.php:84]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie/Misc.php [hunk 10]
```diff
@@ -2194,8 +2181,29 @@ function embed_wmedia(width, height, link) {
 	 */
 	public static function get_build()
 	{
-		$mtime = @filemtime(dirname(dirname(__FILE__)) . '/SimplePie.php');	//FreshRSS
```
Comment (math-GH) [inline lib/SimplePie/SimplePie/Misc.php:2197]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie/Parser.php [hunk 11]
```diff
@@ -76,86 +76,69 @@ public function set_registry(SimplePie_Registry $registry)
...
+			$data = substr($data, 2);
+		}
+		// UTF-8 BOM
+		elseif (substr($data, 0, 3) === "\xEF\xBB\xBF")
+		{
+			$data = substr($data, 3);
 		}
 
-		if ($xmlEncoding === '' || $xmlEncoding === 'UTF-8')	//FreshRSS: case of no explicit HTTP encoding, and lax UTF-8
+		if (substr($data, 0, 5) === '<?xml' && strspn(substr($data, 5, 1), "\x09\x0A\x0D\x20") && ($pos = strpos($data, '?>')) !== false)
```
Comment (math-GH) [inline lib/SimplePie/SimplePie/Parser.php:130]: still needed? Maybe we should add more information why we need it

File: lib/SimplePie/SimplePie/Sanitize.php [hunk 12]
```diff
@@ -318,7 +319,6 @@ public function sanitize($data, $type, $base = '')
 		{
 			if ($type & SIMPLEPIE_CONSTRUCT_MAYBE_HTML)
 			{
-				$data = htmlspecialchars_decode($data, ENT_QUOTES);	//FreshRSS
```
Comment (math-GH) [inline lib/SimplePie/SimplePie/Sanitize.php:321]: still needed? Maybe we should add more information why we need it

**Comment by Alkarex** [comment]
No, sorry. Our local library has several patches. The way to go is to patch upstream, which I already have done to a good extent https://github.com/simplepie/simplepie/pulls?q=is%3Apr+author%3AAlkarex , but more work is needed. Some things might require a bit of tuning before upstream acceptance.
What can also be done is to regularly cherry-pick the commits upstream (which I also do regularly, cf. e.g. https://github.com/FreshRSS/FreshRSS/pull/4011 )

**Comment by Alkarex** [comment]
Manual update done in https://github.com/FreshRSS/FreshRSS/pull/4113

**Pull request closed without merging**